pkg/registries/adapters/dockerhub_adapter.go

```diff
@@ -111,8 +111,7 @@ func (r DockerHubAdapter) FetchSpecs(imageNames []string) ([]*apb.Spec, error) {
 	for _, imageName := range imageNames {
 		spec, err := r.loadSpec(imageName)
 		if err != nil {
-			r.Log.Errorf("unable to retrieve spec data for image - %v", err)
-			return specs, err
+			r.Log.Errorf("Failed to retrieve spec data for image %s - %v", imageName, err)
```

Comment:
I personally don't like this approach.

This needs to be changed on the interface, if we are never going to return something went wrong I think we should force that across all adapters IMO. I think this is a conversation that we should have about what we expect from the registry perspective if an adapter throws an error here. Do we want to respect it? do we want the adapter to handle its own errors and just pass us a clean list of specs?

Reply:
The problem with moving out to the interface is with the local_openshift_adapter unfortunately. It has a few places where we need to respect the errors for, like creating the openshift client, checking the ip of the docker-registry, and loading images from the docker-registry.  

> I think this is a conversation that we should have about what we expect from the registry perspective if an adapter throws an error here. Do we want to respect it? do we want the adapter to handle its own errors and just pass us a clean list of specs?
>
I think it's worth having.  In the near term though (3.9), this PR will ignore any failed specs which will prevent anyone from hitting the issue. I think it's worth coming back to do more thorough improvement on this.  We can do things like log how many bad specs were found in a repo and define a firmer policy on handling registry errors.

Reply:
👍 that sounds reasonable to me